test/correctness/fast_function_approximations.cpp

```diff
@@ -0,0 +1,649 @@
+#include "Halide.h"
+
+#include <cinttypes>
+#include <cmath>
+#include <cstdio>
+#include <locale.h>
+#include <string>
+
+using namespace Halide;
+using namespace Halide::Internal;
+
+const bool use_icons = true;
```

Comment:
Get rid of this. If you want icons, print them for each subcase and then print a single unambiguous piece of text for the set that can be searched for via grep or in an editor. (The text is the worst case result.) Colored test would also be fine, but requires terminal support. The icons have already wasted my time as I can't search for them easily and there are a lot of test cases in the test. A flag that one edits in the source code for this is just silly. Yeah, we ought to have a library for writing tests, but we don't. Keep it simple.

Reply:
I agree with @zvookin. No information should be emoji-only. Emojis shouldn't be sent to terminals that don't support them (e.g. on Windows, I think you need to set the code page). Let's avoid introducing emojis to the codebase in this PR.

Reply:
Can I just default the bool to false? I found it very useful to have these emojis. Color is a very good dimension of information when you're skimming through the many results.